fplc_sample_manager.py

```
@@ -0,0 +1,489 @@
+            logger.error(f"Attempted to register duplicate sample ID: {sample_id}")
+            raise DuplicateSampleError(f"Sample with ID '{sample_id}' already exists.")
+
+        try:
+            new_sample = Sample(sample_id=sample_id, metadata=metadata)
+            self._samples[sample_id] = new_sample
+            self._save_samples()
+            logger.info(f"Sample '{sample_id}' registered successfully.")
+            return new_sample
+        except ValueError as e:
+            logger.error(f"Validation error during sample registration for '{sample_id}': {e}")
+            raise SampleManagerError(f"Failed to register sample '{sample_id}': {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)